### PR TITLE
Creating a homepage

### PR DESCRIPTION
**Pull request by JohnAsare**
A pull request for master

### CODE REVIEW

**Comment by JohnAsare** [comment]
Made a pull request

**Comment by nctl144** [comment]
@JohnAsare if you are using other text editors (PyCharm), it is generating some unexpected files. Therefore, you will need to delete them before you commit

File: .idea/vcs.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
```
Comment (nctl144) [inline .idea/vcs.xml]: This is an unexpected file and should not be included in the Pull Request

**Comment by nctl144** [comment]
@JohnAsare for this PR you will need to define an extra function in **views.py** that leads to your template that you made in this PR. Something like this will work:

```
def homepage(request):
    return render(request, 'home/[name-of-template].html')
```

In addition, you will need to define the url path that this template will be available to be displayed. This can be done in urls.py. Add something like this to the urlpatterns array:
`path('homepage/', views.homepage, name='homepage'),`

Then you should see the template that you created being served at http://127.0.0.1:8000/homepage.

Let me know if you have any other question.

File: home/templates/home/Try.html [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    var x = document.getElementById("navDemo");
+    if (x.className.indexOf("w3-show") == -1) {
+        x.className += " w3-show";
+    } else {
+        x.className = x.className.replace(" w3-show", "");
+    }
+}
+</script>
+
+</body>
```
Comment (nctl144) [inline home/templates/home/Try.html]: for Django, you wont need to include <body> and <html> tags since those tags are going to be generated

File: home/templates/home/Try.html [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+</html>
```
Comment (nctl144) [inline home/templates/home/Try.html]: Same for this tag

File: home/templates/home/Try.html [hunk 4]
```diff
@@ -0,0 +1,137 @@
+<!DOCTYPE html>
```
Comment (nctl144) [inline home/templates/home/Try.html]: This is also not needed since Django will generate the tag

File: home/templates/home/Try.html [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+<html>
+<title>W3.CSS Template</title>
+<meta charset="UTF-8">
+<meta name="viewport" content="width=device-width, initial-scale=1">
```
Comment (nctl144) [inline home/templates/home/Try.html]: this tag is not needed

File: home/templates/home/Try.html [hunk 6]
```diff
@@ -0,0 +1,137 @@
+<!DOCTYPE html>
+<html>
+<title>W3.CSS Template</title>
+<meta charset="UTF-8">
```
Comment (nctl144) [inline home/templates/home/Try.html]: This is also not needed

File: home/templates/home/Try.html [hunk 7]
```diff
@@ -0,0 +1,137 @@
+<!DOCTYPE html>
+<html>
```
Comment (nctl144) [inline home/templates/home/Try.html]: this tag is not needed

File: home/templates/home/Try.html [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+<title>W3.CSS Template</title>
```
Comment (nctl144) [inline home/templates/home/Try.html]: remove the title for now

File: home/templates/home/Try.html [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+<meta charset="UTF-8">
+<meta name="viewport" content="width=device-width, initial-scale=1">
+<link rel="stylesheet" href="https://www.w3schools.com/w3css/4/w3.css">
```
Comment (nctl144) [inline home/templates/home/Try.html]: Check the Intro.html file. **{% load static %}** has to be put at the beginning of each template. Copy the <link> from the Intro.html as well as the scripts from Intro.html as they are the essential css and script for each template of the project

File: .idea/vcs.xml (hunk 1 above)
Comment (nctl144) [reply .idea/vcs.xml]: @JohnAsare remember to delete the **.idea** folder that the text editor generated

**Comment by nctl144** [comment]
I didnt see you connect the template with the **urls.py** and **views.py** @JohnAsare

File: home/views.py [hunk 10]
```diff
@@ -5,6 +5,7 @@
 from .models import Product
 from .forms import SearchForm
 
+
```
Comment (nctl144) [inline home/views.py]: Can you delete this line? When you work on your PR, just concentrate on writing the code that you need to write. Other than that, don't change anything :)

File: home/views.py [hunk 11]
```diff
@@ -22,6 +23,7 @@ def index(request):
     # products = paginator.get_page(page)
     return render(request, 'home/index.html', {'form': form})
 
+
```
Comment (nctl144) [inline home/views.py]: same for this line

File: home/views.py [hunk 12]
```diff
@@ -33,3 +35,19 @@ def result(request):
...
+    keywords = ''
+    if request.method == 'POST':
+        form = SearchForm(request.POST)
+        if form.is_valid():
+            keywords = form.cleaned_data['Keywords']
+
+    else:
+        form = SearchForm()
+
+    return render(request, 'home/intro.html', {'form': form, 'keywords': keywords})
```
Comment (nctl144) [inline home/views.py]: You just need return render(request, 'home/intro.html') for this function, delete everything except for that.

File: home/static/home/css/main.css [hunk 13]
```diff
@@ -121,6 +121,8 @@ span.label-info span {
     margin: 0 auto;
 }
 
+/*@Lam this is the change I made */
+
```
Comment (nctl144) [inline home/static/home/css/main.css]: You don't really need to write this as I know what you addded

File: home/views.py [hunk 14]
```diff
@@ -33,3 +35,19 @@ def result(request):
         form = SearchForm()
 
     return render(request, 'home/result.html', {'form': form, 'keywords': keywords})
+
+
+# @Lam this is the change I made.
```
Comment (nctl144) [inline home/views.py]: Delete this
